### ISSUE TITLE
The dependency target "omptarget.devicertl" of target  "omptarget.rtl.x86_64.unittests" does not exist

### ISSUE THREAD

**Issue by kwk** [llvm/llvm-project#76225]
I see this error when building 7c3b67d2038cfb.

```
CMake Error at /builddir/build/BUILD/llvm-project-18.0.0.src/openmp/libomptarget/unittests/Plugins/CMakeLists.txt:8 (add_dependencies):
  The dependency target "omptarget.devicertl" of target
  "omptarget.rtl.x86_64.unittests" does not exist.
```

(See https://github.com/kwk/llvm-daily-fedora-rpms/issues/142 or https://download.copr.fedorainfracloud.org/results/@fedora-llvm-team/llvm-snapshots-big-merge-20231222/fedora-38-x86_64/06782394-llvm/builder-live.log.gz)

That means for us the error from https://github.com/llvm/llvm-project/pull/74398 has shifted and while it might be fixed by https://github.com/llvm/llvm-project/pull/76141, we're stuck again.

**Comment by kwk** [comment]
A temporary and probably dirty fix to get past this CMake error is this:

```diff
--- /openmp/libomptarget/unittests/Plugins/CMakeLists.txt.old	2023-12-22 16:41:39.382052043 +0000
+++ /openmp/libomptarget/unittests/Plugins/CMakeLists.txt	2023-12-22 16:33:17.934052043 +0000
@@ -5,7 +5,9 @@
 foreach(PLUGIN IN LISTS LIBOMPTARGET_TESTED_PLUGINS)
   libomptarget_say("Building plugin unit tests for ${PLUGIN}")
   add_libompt_unittest("${PLUGIN}.unittests" ${PLUGINS_TEST_SOURCES})
-  add_dependencies("${PLUGIN}.unittests" ${PLUGINS_TEST_COMMON} ${PLUGIN})
-  target_link_libraries("${PLUGIN}.unittests" PRIVATE ${PLUGINS_TEST_COMMON} ${PLUGIN})
-  target_include_directories("${PLUGIN}.unittests" PRIVATE ${PLUGINS_TEST_INCLUDE})
+  if(NOT PLUGIN STREQUAL "omptarget.rtl.x86_64")
+    add_dependencies("${PLUGIN}.unittests" ${PLUGINS_TEST_COMMON} ${PLUGIN})
+    target_link_libraries("${PLUGIN}.unittests" PRIVATE ${PLUGINS_TEST_COMMON} ${PLUGIN})
+    target_include_directories("${PLUGIN}.unittests" PRIVATE ${PLUGINS_TEST_INCLUDE})
+  endif()
 endforeach()
```

**Comment by kwk** [comment]
Thank you @EthanLuisMcDonough for the fix in #76266. The link is currently not available but it will be tomorrow with results the Fedora 38 x86_64: https://copr.fedorainfracloud.org/coprs/g/fedora-llvm-team/llvm-snapshots-big-merge-20231223/monitor/ . Only then I can confirm if it works or not.
